src/io/gff.rs

```diff
@@ -91,28 +143,39 @@ impl<'a, R: io::Read> Iterator for Records<'a, R> {
 /// A GFF writer.
 pub struct Writer<W: io::Write> {
     inner: csv::Writer<W>,
+    gff_type: GffType,
 }
 
 
 impl Writer<fs::File> {
     /// Write to a given file path.
-    pub fn to_file<P: AsRef<Path>>(path: P) -> io::Result<Self> {
-        fs::File::create(path).map(Writer::new)
+    pub fn to_file<P: AsRef<Path>>(path: P, fileformat: GffType) -> io::Result<Self> {
+        fs::File::open(path).map(|f| Writer::new(f, fileformat))
     }
 }
 
 
 impl<W: io::Write> Writer<W> {
     /// Write to a given writer.
-    pub fn new(writer: W) -> Self {
-        Writer { inner: csv::Writer::from_writer(writer).delimiter(b'\t').flexible(true) }
+    pub fn new(writer: W, fileformat: GffType) -> Self {
+        Writer {
+            inner: csv::Writer::from_writer(writer).delimiter(b'\t').flexible(true),
+            gff_type: fileformat,
+        }
     }
 
     /// Write a given GFF record.
     pub fn write(&mut self, record: Record) -> csv::Result<()> {
+        let (delim, termi) = self.gff_type.get_separator();
+
+        // If you can remove this ugly code please do it
+        let termi = [termi];
+        let termi = str::from_utf8(&termi).unwrap_or(";");
+
         let attributes;
+
         if !record.attributes.is_empty() {
-            attributes = record.attributes.iter().map(|(a, b)| format!("{}={}", a, b)).collect::<Vec<_>>().join(";");
+            attributes = record.attributes.iter().map(|(a, b)| format!("{}{}{}", a, delim as char, b)).collect::<Vec<_>>().join(&termi);
```

Comment:
You can use the join method from itertools here. This avoids the allocation to a vector.